### PR TITLE
Add a template for custom 404 page

### PR DESCRIPTION
**Pull request by ashmaroli**
Ref: #5944 

Provide a Custom 404 Error Page with `jekyll new`

Personally, not very happy about including internal CSS. However, this is preferable until Minima decides to ship a layout dedicated to this page.

/cc @pathawks @benbalter

### CODE REVIEW

**Comment by crispgm** [comment]
It's a good idea. However, it may hardly take effect because we haven't taken control of webserver.

**Comment by ashmaroli** [comment]
This works on `localhost:4000`
This should work on GH Pages as per [this Help Article](https://help.github.com/articles/creating-a-custom-404-page-for-your-github-pages-site/)

For Apache webservers, users may need to modify their `.htaccess` but that's a different story.

**Comment by ashmaroli** [comment]
A Custom 404 page on the WEBrick development server

**Comment by ashmaroli** [comment]
It does work on GH Pages. [Sample this](https://ashmaroli.github.io/test-site/aloha)

**Comment by pathawks** [comment]
Thanks @ashmaroli! Is this documented in _our_ docs? If not, it should be :beers:

**Comment by ashmaroli** [comment]
> Is this documented..

Pardon me, but what exactly were you pointing at?

**Comment by crispgm** [comment]
@ashmaroli Looks cool! I think we should add a new section on docs to show how to make it work with webservers like `nginx`.

**Comment by ashmaroli** [comment]
> I think we should add a new section on docs to show how to make it work with webservers like nginx.

Since this works out-of-the-box on GitHub Pages, I feel *a how to..* documentation for `nginx` and `Apache` softwares should be on a separate PR to the `tutorials` collection.

File: lib/site_template/404.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    margin: 10px auto;
+    width: 600px;
+    text-align: center;
+  }
+  h1 {
+    width: 800px;
+    position: relative;
+    left: -100px;
+    margin: 30px 0;
+    font-size: 60px;
```
Comment (DirtyF) [inline lib/site_template/404.html]: One commonly used blog template should be mobile-first and responsive, even here.

Comment (ashmaroli) [inline lib/site_template/404.html]: Thnx @DirtyF I had totally ignored mobile-friendliness.. 
Will update.

Comment (ashmaroli) [inline lib/site_template/404.html]: updated.

404m

File: lib/site_template/404.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  }
+</style>
+
+<div class="container">
+  <h1>404</h1>
+
+  <p><strong>Page not found :(</strong></p>
+  <p>The requested page could not be found.</p>
+
+  <p>Edit the <code>404.html</code> file in your site to show a custom 404 Error Message.</p>
```
Comment (benbalter) [inline lib/site_template/404.html]: Perhaps, "If you're the maintainer of the site, edit the..."

Comment (ashmaroli) [inline lib/site_template/404.html]: I intended it to be similar to the imperative voice in default `{{ site.description }}`

Comment (benbalter) [inline lib/site_template/404.html]: My thinking was that it's almost assumed that you're going to change the default description, but that we shouldn't *require* a site publisher to customize the 404 (either due to the imperative voice or to remove something that looks potentially embarrassing).

Comment (ashmaroli) [inline lib/site_template/404.html]: Valid point. I'll just remove that line entirely.

**Comment by parkr** [comment]
Neat!

@jekyllbot: merge +minor

**Pull request merged**